src/SFA.DAS.ApprenticeCommitments/Extensions/DbContextBuilderExtensions.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Data.Common;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+
+namespace SFA.DAS.ApprenticeCommitments.Extensions
+{
+    public static class DbContextBuilderExtensions
+    {
+        public static DbContextOptionsBuilder<TContext> UseDataStorage<TContext>(this DbContextOptionsBuilder<TContext> builder, IConfiguration config, string connection) where TContext : DbContext
+        {
+            if (config.IsAcceptanceTest())
+            {
+                return builder.UseSqlite(connection);
```

Comment:
I'm not convinced about putting SQLite into the main code.  The [MS docs suggest](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-3.1#customize-webapplicationfactory) replacing the DbContext in the inherited `WebApplicationFactory`.  Do you think that might work in this case?

Reply:
yes absolutely agree, I think good way would be to introduce a dataStorage factory which would create a sqlconnection and then in the tests we replace this with a SOLite connection. This should keep SQL lite out of our code

Reply:
this is now fixed

Reply:
That `ConnectionFactory` separates the tests from the main code beautifully 💯

Reply:
yes i was really pleased with it. it had bothered me how I'd first implemented so i was glad you picked it up